salt/modules/linux_shadow.py

```diff
@@ -37,9 +37,11 @@
 except ImportError:
     HAS_CRYPT = False
 
+__virtualname__ = "shadow"
```

Comment:
Should we also be renaming this file?

Reply:
And I guess the `linux_service.py`

Reply:
As in, is it appropriate to mv shadow to linux_shadow and service to linux_service? Yes, they have to be updated. See the previous comments for more context.

From above:
> Added a test to check each module's rst for the text ".. _virtual" which successfully detects problems with shadow and service execution module naming.
> With this additional check, the doc test is expected to fail based on the mentioned issues with shadow and service. I don't think correcting those should be part of his PR, so waiting for input from core team on next steps. I suggest merging #53296 and an equivalent for service.

(The referenced PR was since merged into this one to get the mentioned tests passing)

The issue is #49335 and also #52000. Salt docs generate index pages for virtual modules. Since service is a virtual module name, having a module also named service means that any links to the service module, instead are directed to the virtual module index page. This means that on 
https://docs.saltstack.com/en/2018.3/ref/modules/all/salt.modules.service.html you are unable to click on the service module link name in the middle, as it just loads the current page rather than the module doc. 

This was a problem for the generically named shadow and service modules. A test was added in this PR to detect modules which duplicate virtual module index names, so they had to be renamed in order to keep tests passing.

Reply:
Okay, I'm pretty sure I'm down with that. If we're renaming things only *within* salt, but they're not part of the API, I think it's a great idea to name them consistently.

If it borked with the public API then we'd have to deprecate things, but if people shouldn't be importing this code themselves then I'm :100:

Reply:
As far as I know, module file names are not part of the salt api. Advice has always been to call other modules via \_\_salt\_\_['whatever'] rather than direct imports (https://docs.saltstack.com/en/latest/ref/modules/index.html#cross-calling-execution-modules).

Feel free to discuss with the everyone, but as far as I know, this should be a-ok!